### PR TITLE
Auth: Optimize auth token operations

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

- Add method for counting active tokens (system wide and user specific)

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/auth/authimpl/auth_token.go [hunk 1]
```diff
@@ -553,22 +570,16 @@ func (s *UserAuthTokenService) GetUserRevokedTokens(ctx context.Context, userId
 }
 
 func (s *UserAuthTokenService) reportActiveTokenCount(ctx context.Context, _ *quota.ScopeParameters) (*quota.Map, error) {
-	var count int64
```
Comment (eleijonmarck) [inline pkg/services/auth/authimpl/auth_token.go:556]: do we know that this is more performant? if so why? 🤔

Comment (Jguer) [reply pkg/services/auth/authimpl/auth_token.go:556]: check the sister PR to this one, it stops us from retrieving all tokens just to len them

Comment (eleijonmarck) [reply pkg/services/auth/authimpl/auth_token.go:556]: ah, you mean only the int and not the objects

**Review by gamab — APPROVED** [review]

File: pkg/services/auth/authtest/testing.go [hunk 2]
```diff
@@ -20,12 +20,12 @@ type FakeUserAuthTokenService struct {
...
 	RevokeTokenProvider          func(ctx context.Context, token *auth.UserToken, soft bool) error
-	RevokeAllUserTokensProvider  func(ctx context.Context, userId int64) error
-	ActiveAuthTokenCount         func(ctx context.Context) (int64, error)
-	GetUserTokenProvider         func(ctx context.Context, userId, userTokenId int64) (*auth.UserToken, error)
-	GetUserTokensProvider        func(ctx context.Context, userId int64) ([]*auth.UserToken, error)
-	GetUserRevokedTokensProvider func(ctx context.Context, userId int64) ([]*auth.UserToken, error)
-	BatchRevokedTokenProvider    func(ctx context.Context, userIds []int64) error
+	RevokeAllUserTokensProvider  func(ctx context.Context, userID int64) error
+	ActiveTokenCountProvider     func(ctx context.Context, userID *int64) (int64, error)
+	GetUserTokenProvider         func(ctx context.Context, userID, userTokenId int64) (*auth.UserToken, error)
```
Comment (mgyongyosi) [inline pkg/services/auth/authtest/testing.go:25]: nit:
```suggestion
	GetUserTokenProvider         func(ctx context.Context, userID, userTokenID int64) (*auth.UserToken, error)
```

**Review by mgyongyosi — APPROVED** [review]

**Comment by eleijonmarck** [comment]
manually backporting this

```bash
# Fetch latest updates from GitHub
git fetch
# Create a new branch
git switch --create backport-74602-to-v10.1.x origin/v10.1.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick -x 77e4d477e579ce1204e5a2967e1051f880398004
# When the conflicts are resolved, stage and commit the changes
git add . && git cherry-pick --continue
```

**Comment by eleijonmarck** [comment]
creating the PR
```bash
# Create the PR body template
PR_BODY=$(gh pr view 74602 --json body --template 'Backport 77e4d477e579ce1204e5a2967e1051f880398004 from #74602{{ "\n\n---\n\n" }}{{ index . "body" }}')
# Push the branch to GitHub and a PR
echo "${PR_BODY}" | gh pr create --title "[v10.1.x] Auth: Optimize auth token operations" --body-file - --label "type/bug" --label "area/backend" --label "area/backend/db/migration" --label "no-changelog" --label "backport" --base v10.1.x --milestone 10.1.x --web
```

**Pull request merged**